### PR TITLE
Fixes so that the app does not crash on iPad 

### PR DESCRIPTION
**Pull request by joshmcarthur**
This pull request hopefully gives us a working release that the iTunes team will approve for beta release, unblocking our testing!

After I updated the UI and workflow of the phone version of the app, the iPad layout was left in a bit of a broken state. This pull request updates the behaviour of the iPad app so that the app can be opened and used. I've made some design modifications to get it looking more like the phone app, but functionality-aside, design changes are out of scope for this phase of work, so I've not done more than I need to to get the app unbroken. I'm hoping to have some time this week to jump in and try and make some more improvements, as I know many families use this app on an iPad due to it's larger size and readability.

simulator screen shot 12 06 2017 3 44 09 pm

simulator screen shot 12 06 2017 3 44 16 pm

There's a BUNCH of things annoying me here that I'd like to fix (but don't affect the app working)

1. The blue on the search bar is  darker. There's no reason in code for it. It just is. ???
2. I'd like to fix the webview containing the about content to the bottom of the screen. I also wonder if it should go in the right hand column when there is no search (I think this is the 'out of scope' bit)
3. The layout of the main part of the screen needs some major typographical and spacing adjustments to look nicer
4. I'd like to put the "NZSL Dictionary" app title up the top left like the phone version has. It's going to be a bit of a positioning nightmare.

### CODE REVIEW

File: NZSLDict/Classes/AppDelegate.swift [hunk 1]
```diff
@@ -13,6 +13,9 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             rootViewController = ViewControllerPhone()
         } else {
             rootViewController = ViewControllerPad()
+            self.window!.rootViewController = rootViewController
+            self.window!.makeKeyAndVisible();
```
Comment (Br3nda) [inline NZSLDict/Classes/AppDelegate.swift]: wow this is very ruby

**Comment by Br3nda** [comment]
Some CI would be choice as.

File: NZSLDict/Classes/DetailViewController.swift [hunk 2]
```diff
@@ -33,12 +32,18 @@ class DetailViewController: UIViewController, UISplitViewControllerDelegate, UIN
...
         navigationBar.autoresizingMask = .FlexibleWidth
+        navigationBar.titleTextAttributes = [NSForegroundColorAttributeName: UIColor.whiteColor()]
         navigationBar.delegate = self
         view.addSubview(navigationBar)
+        
+        navigationTitle = UINavigationItem(title: "")
+        navigationBar.setItems([navigationTitle], animated: false)
 
-        diagramView = DiagramView(frame: CGRectMake(0, top_offset + 44, view.bounds.size.width, view.bounds.size.height / 2))
+        diagramView = DiagramView(frame: CGRectMake(0, navigationBar.frame.maxY, view.bounds.size.width, view.bounds.size.height / 2))
```
Comment (Br3nda) [inline NZSLDict/Classes/DetailViewController.swift]: this is nicer!

File: NZSLDict/Classes/SearchViewController.swift [hunk 3]
```diff
@@ -350,7 +373,7 @@ class SearchViewController: UIViewController, UISearchBarDelegate, UITableViewDa
         if sender.state == .Ended {
             self.selectEntry(wordOfTheDay)
             searchBar.resignFirstResponder()
-            self.delegate.didSelectEntry(wordOfTheDay)
+            self.delegate?.didSelectEntry(wordOfTheDay)
```
Comment (Br3nda) [inline NZSLDict/Classes/SearchViewController.swift]: what's the `?` doing for us? Is it just a method name like in ruby -- or something magic?

File: NZSLDict/Classes/SearchViewController.swift [hunk 4]
```diff
@@ -398,7 +421,7 @@ class SearchViewController: UIViewController, UISearchBarDelegate, UITableViewDa
     }
 
     func selectEntry(entry: DictEntry) {
-        NSNotificationCenter.defaultCenter().postNotificationName(EntrySelectedName, object: self, userInfo: ["entry": entry])
+        NSNotificationCenter.defaultCenter().postNotificationName(EntrySelectedName, object: nil, userInfo: ["entry": entry])
```
Comment (Br3nda) [inline NZSLDict/Classes/SearchViewController.swift]: why does the object need to be passed here?

**Comment by Br3nda** [comment]
Do you think we should add Raygun so we have more info on strange crashes that might get reported.

File: NZSLDict/Classes/SearchViewController.swift (hunk 3 above)
Comment (joshmcarthur) [inline NZSLDict/Classes/SearchViewController.swift]: @Br3nda it's magic - it tells Swift not to throw a runtime error if `self.delegate` is `nil`. The delegate is nil in the iPad view.

File: NZSLDict/Classes/SearchViewController.swift (hunk 4 above)
Comment (joshmcarthur) [inline NZSLDict/Classes/SearchViewController.swift]: @Br3nda my understanding is that the arguments of `postNotificationName` must match what the receiver is expecting for the notification to be triggered. With `object: self`, the received notification had no `userInfo` set, whereas with `object: nil`, the `userInfo` was present.

Like many of these changes, this is just the result of me poking the code until it does what I want.

**Comment by joshmcarthur** [comment]
@Br3nda RE: CI and Raygun. I'd like to add these, but I'd like to get through the changes that Micky has asked for first (same goes for Android). In my experience, setting up CI for a native project is not as simple as a web application, and (especially so for iOS) I expect to need to do a lot of research and debugging in Travis before the CI would be up and running.

**Pull request merged**